File: j2k-eval/edge-cases/src/EnumWithBehavior.java
package edgecases;
public enum EnumWithBehavior {
    ADD("+") { @Override public double apply(double a, double b) { return a + b; } },
    SUBTRACT("-") { @Override public double apply(double a, double b) { return a - b; } },
    MULTIPLY("*") { @Override public double apply(double a, double b) { return a * b; } },
    DIVIDE("/") { @Override public double apply(double a, double b) { if (b == 0) throw new ArithmeticException("Division by zero"); return a / b; } };
    private final String symbol;
    EnumWithBehavior(String symbol) { this.symbol = symbol; }
    public String getSymbol() { return symbol; }
    public abstract double apply(double a, double b);
    public static EnumWithBehavior fromSymbol(String symbol) { for (EnumWithBehavior op : values()) { if (op.symbol.equals(symbol)) return op; } throw new IllegalArgumentException("Unknown: " + symbol); }
}

File: j2k-eval/edge-cases/converted/EnumWithBehavior.kt
package edgecases

/**
 * Edge case: Enums with abstract methods, fields, and complex behavior.
 * Hypothesis: The converter may struggle with enum constants that override methods.
 */
enum class EnumWithBehavior(val symbol: String) {
    ADD("+") {
        override fun apply(a: Double, b: Double): Double {
            return a + b
        }
    },
    SUBTRACT("-") {
        override fun apply(a: Double, b: Double): Double {
            return a - b
        }
    },
    MULTIPLY("*") {
        override fun apply(a: Double, b: Double): Double {
            return a * b
        }
    },
    DIVIDE("/") {
        override fun apply(a: Double, b: Double): Double {
            if (b == 0.0) throw java.lang.ArithmeticException("Division by zero")
            return a / b
        }
    };

    abstract fun apply(a: Double, b: Double): Double

    companion object {
        // Static factory method
        fun fromSymbol(symbol: String): EnumWithBehavior {
            for (op in entries) {
                if (op.symbol == symbol) return op
            }
            throw java.lang.IllegalArgumentException("Unknown operator: $symbol")
        }
    }
}